vdev/src/app.rs

```diff
@@ -58,6 +59,27 @@ pub fn version() -> Result<String> {
         if tag != format!("v{version}") {
             bail!("On latest release channel and tag {tag:?} is different from Cargo.toml {version:?}. Aborting");
         }
+    } else if channel == "custom" {
+        let base_version = util::read_version()?;
+
+        // use pre computed SHA if available, such as when called within build workflows,
```

Comment:
Ah I see. Do you have a link to one of these failures? I'm a bit confused as to what would cause it.

Reply:
Mhm. Basically it's a git permissions thing that happens within containers. We have a command run in the publish workflow to handle it for some of those cases, but in this failure it's because it is failing from the call out to the Makefile targets that do the packaging, which invokes the docker run, hence my changes to pass forward that env var in the Makefile.

https://github.com/vectordotdev/vector/actions/runs/4428339593/jobs/7767355987

Reply:
My comment in the `build.rs` file actually does a little better job explaining the same reasoning that this one is trying to do. I can transpose that one here.

Reply:
We do actually have a precedent of setting `safe.directory` to workaround that issue: https://github.com/vectordotdev/vector/blob/master/scripts/environment/prepare.sh#L4

Maybe we could do that here too? Or call that same prepare.sh script?

Reply:
Good idea. I was working around the issue by adding the sha to an env var. But this seems like a better approach. My first attempt was to add the safe directory in the workflow job, but that didn't work since the issue was in the container.
But we can still run that command while we are in the scripts executing within the container.

Reply:
@jszwedko - one thing I just thought of as I'm implementing this. Regardless of whether we compute the SHA and store in a env var, or mark the repo as safe, it strikes me that this could introduce a problem for anyone who is building vector from source, within a container. I guess the only way around that would be to add logic into `build.rs` , which would call the git command to add safe directory, for the current working directory.

Do you think this is a valid concern,?

Reply:
The other comment raised about explicitly throwing error if the git command fails, is essentially what would otherwise happen, if we didn't do anything.

Reply:
We discussed offline and decided that this is a marginal concern and that it is reasonable to expect the permissions to be set, but I will comment it.